tokio/src/sync/watch.rs

```diff
@@ -433,31 +434,51 @@ impl<T> Sender<T> {
             return Err(error::SendError(value));
         }
 
-        self.send_replace(value);
```

Comment:
If the destructor of the old value panics, then the new value is not sent. We should run the destructor after the `send_modify` call instead to keep the old behavior.